src/passes/pass.cpp

```diff
@@ -546,27 +546,33 @@ static void dumpWast(Name name, Module* wasm) {
 
 void PassRunner::run() {
   static const int passDebug = getPassDebug();
-  if (!isNested && (options.debug || passDebug)) {
+  // Emit logging information when asked for. At passDebug level 1+ we log
+  // the main passes, while in 2 we also log nested ones. Note that for
+  // nested ones we can only emit their name - we can't validate, or save the
+  // file, or print, as the wasm may be in an intermediate state that is not
+  // valid.
+  if (options.debug || (passDebug == 2 || (passDebug && !isNested))) {
     // for debug logging purposes, run each pass in full before running the
     // other
     auto totalTime = std::chrono::duration<double>(0);
-    size_t padding = 0;
     WasmValidator::Flags validationFlags = WasmValidator::Minimal;
     if (options.validateGlobally) {
       validationFlags = validationFlags | WasmValidator::Globally;
     }
-    std::cerr << "[PassRunner] running passes..." << std::endl;
+    auto what = isNested ? "nested passes" : "passes";
```

Comment:
```suggestion
    auto what = isNested ? "  nested passes" : "passes";
```
We talked offline and it is hard to track depths of nested passes, but maybe it'd be good to just add a uniform indentation for nested passes?

Reply:
That would be good, but I think this change would just make it print `Running    nested passes` (that is, the extra indentation is not at the beginning of the line).

Reply:
Looking around, it doesn't seem like there is an easy and obvious way to add indentation here (it would need to affect more than one place), so let's leave that for work later maybe.